src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -351,6 +357,20 @@ internal static async Task ProcessSarifLogAsync(Stream stream, bool showMessageO
             if (sarifLog != null)
             {
                 await ProcessSarifLogAsync(sarifLog, logFilePath: null, showMessageOnNoResults: showMessageOnNoResults, cleanErrors: cleanErrors, openInEditor: openInEditor);
+                if (sarifLog.HasNoResults())
```

Comment:
This is an example of where I would expect ProcessSarifLogAsync to return something like Task<ProcessLogResults> which would contain information that is being inspected here. The caller of this should be doing UI work.  I.E Let's not do UI stuff directly in the error service

Reply:
Agreed. The actual logic for deciding which bars to show (both in the single-log and multiple-log case) is non-trivial, and easily separable and unit testable, so I'm going to refactor to that. This will help address Eddy's question (can you have no results _and_ a configuration error? the answer is "yes") in a way that will be easy to see.

I propose:
- If a given log file has a catastrophic error, do _not_ display the "no results" bar, regardless of whether the log contains results.
- Across multiple logs, display each "kind" of bar only once.

The result is that you will see at most one _each_ of "no results", "catastrophic configuration error", "catastrophic execution error", and "invalid SARIF file" across any set of logs.

---
In reply to: [523438595](https://github.com/microsoft/sarif-visualstudio-extension/pull/272#discussion_r523438595) [](ancestors = 523438595)

Reply:
This is now done.

---
In reply to: [523445509](https://github.com/microsoft/sarif-visualstudio-extension/pull/272#discussion_r523445509) [](ancestors = 523445509,523438595)